operator/controller/src/test/java/io/apicurio/registry/operator/it/TlsITTest.java

```diff
@@ -0,0 +1,162 @@
+            assertThat(client.network().v1().ingresses().inNamespace(namespace)
+                    .withName(registry.getMetadata().getName() + "-ui-ingress").get().getSpec().getRules()
+                    .get(0).getHost()).isEqualTo(registry.getSpec().getUi().getIngress().getHost());
+            return true;
+        });
+
+        // Network Policy
+        await().ignoreExceptions().until(() -> {
+            assertThat(client.network().v1().networkPolicies().inNamespace(namespace)
+                    .withName("simple-app-networkpolicy").get().getSpec().getIngress()
+                    .get(0).getPorts().get(0).getPort().getIntVal()).isEqualTo(8443);
+            return true;
+        });
+    }
+
+    /**
+     * In this test, the server uses a Keystore in PCKS format that identifies itself to the client.
+     */
+    @Test
+    void testTLSInsecureTrafficEnabled() {
+        var registry = ResourceFactory.deserialize("/k8s/examples/tls/insecure-traffic-with_tls.apicurioregistry3.yaml",
+                ApicurioRegistry3.class);
+        registry.getMetadata().setNamespace(namespace);
+        registry.getSpec().getApp().getIngress().setHost(ingressManager.getIngressHost("app"));
+        registry.getSpec().getUi().getIngress().setHost(ingressManager.getIngressHost("ui"));
+
+        client.resource(registry).create();
+
+        // Deployments
+        await().ignoreExceptions().until(() -> {
+            assertThat(client.apps().deployments().inNamespace(namespace)
+                    .withName(registry.getMetadata().getName() + "-app-deployment").get().getStatus()
+                    .getReadyReplicas()).isEqualTo(1);
+            assertThat(client.apps().deployments().inNamespace(namespace)
+                    .withName(registry.getMetadata().getName() + "-ui-deployment").get().getStatus()
+                    .getReadyReplicas()).isEqualTo(1);
+
+            var appEnv = getContainerFromDeployment(
+                    client.apps().deployments().inNamespace(namespace)
+                            .withName(registry.getMetadata().getName() + "-app-deployment").get(),
+                    REGISTRY_APP_CONTAINER_NAME).getEnv();
+
+            assertThat(appEnv).map(ev -> ev.getName() + "=" + ev.getValue())
+                    .contains(EnvironmentVariables.QUARKUS_HTTP_INSECURE_REQUESTS + "=" + "enabled");
+
+            return true;
+        });
+
+        // Services
+        await().ignoreExceptions().until(() -> {
+            var service = client.services().inNamespace(namespace)
+                    .withName(registry.getMetadata().getName() + "-app-service").get().getSpec();
+
+            assertThat(service.getClusterIP()).isNotBlank();
+            assertThat(service.getPorts().get(0).getPort()).isEqualTo(443);
+            assertThat(service.getPorts().get(1).getPort()).isEqualTo(8080);
+            assertThat(service.getClusterIP()).isNotBlank();
+            return true;
+        });
+
+        // Ingresses
```

Comment:
Since we do not create https Ingresses automatically, this check is probably unnecessary?

Reply:
It is, yes.